### PR TITLE
Support Numpy arrays as seeds in RandomState

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes https://github.com/cupy/cupy/issues/1675

### CODE REVIEW

**Comment by kmaehashi** [comment]
I think it's better to document this behavior somewhere, as this difference can be important for users. Maybe in https://docs-cupy.chainer.org/en/stable/reference/difference.html?

@okuta How do you think about shrinking ndarray seed into 64-bit in this way?

**Comment by okuta** [comment]
OK, This is acceptable.

**Comment by mrocklin** [comment]
I've added a small section to the `difference.html` docs.

**Comment by mrocklin** [comment]
The motivation behind this change is to make it possible for dask.array to work well wth CuPy RandomState objects so that we can generate distributed cupy arrays for future testing.

**Comment by okuta** [comment]
LGTM!

**Comment by okuta** [comment]
jenkins, test this please.

File: cupy/random/generator.py [hunk 1]
```diff
@@ -621,6 +622,8 @@ def seed(self, seed=None):
             except NotImplementedError:
                 seed = numpy.uint64(time.clock() * 1000000)
         else:
+            if isinstance(seed, numpy.ndarray):
+                seed = int(hashlib.md5(seed).hexdigest(), 16) // 2 ** 64
```
Comment (toslunar) [inline cupy/random/generator.py]:
```suggestion
                seed = int(hashlib.md5(seed).hexdigest(), 16) % 2 ** 64
```

**Comment by kmaehashi** [comment]
Jenkins, test this please.

File: cupy/random/generator.py [hunk 2]
```diff
@@ -621,6 +622,8 @@ def seed(self, seed=None):
             except NotImplementedError:
                 seed = numpy.uint64(time.clock() * 1000000)
         else:
+            if isinstance(seed, numpy.ndarray):
+                seed = int(hashlib.md5(seed).hexdigest(), 16) % 2 ** 64
```
Comment (okuta) [inline cupy/random/generator.py]: `[:16]` returns 64bits number.
```suggestion
                seed = int(hashlib.md5(seed).hexdigest()[:16], 16)
```

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
Test failures are not related to this PR.

**Comment by mrocklin** [comment]
Checking in here.  Is there anything I should to do finish this up?

**Comment by okuta** [comment]
Sorry, late reply.
LGTM!

**Pull request merged**